### PR TITLE
feat(text2sql): add a thread pool for copilot

### PR DESCRIPTION
**Pull request by abigsaltyfish**
Put the thread pool under ODC management

### CODE REVIEW

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/config/ScheduleConfiguration.java [hunk 1]
```diff
@@ -317,6 +317,22 @@ public ThreadPoolTaskExecutor queryProfileMonitorExecutor() {
         return executor;
     }
 
+    @Bean(name = "copilotQueryExecutor")
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/config/ScheduleConfiguration.java:320]: 1. why only thread pool, no implementation?

2. if copilot is included in odc, we may add a switch configuration, e.g. odc.features.copilot.enabled

Comment (abigsaltyfish) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/ScheduleConfiguration.java:320]: implementation in odc-cloud, switch configuration will be added after the ODC supports Copilot in the future

Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/ScheduleConfiguration.java:320]: if implemented in odc-cloud, why create thread pool here ?

better add CloudScheduleConfiguration in odc-cloud instead.

Comment (abigsaltyfish) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/ScheduleConfiguration.java:320]: ok

**Pull request closed without merging**